routini/src/set_strategy_endpoint.rs

```
@@ -0,0 +1,88 @@
+use std::sync::Arc;
+
+use http::{Method, Response, StatusCode};
+use pingora::{
+    apps::http_app::ServeHttp, protocols::http::ServerSession, services::listening::Service,
+};
+use serde::de::DeserializeOwned;
+use tracing::{error, info};
+
+use crate::load_balancing::{LoadBalancer, selection::Strategy};
+
+/// Temporary endpoint for updating the load balancer strategy,
+/// This should be automatically decided by an internal task
+#[derive(Clone)]
+pub struct SetStrategyEndpoint<S>
+where
+    S: Strategy + 'static,
+{
+    pub load_balancer: Arc<LoadBalancer<S>>,
+}
+
+impl<S> SetStrategyEndpoint<S>
+where
+    S: Strategy + 'static + DeserializeOwned,
+{
+    pub fn service(load_balancer: Arc<LoadBalancer<S>>, address: &str) -> Service<Self> {
+        let mut service = Service::new("strategy_updater".to_string(), Self { load_balancer });
+        service.add_tcp(address);
+        service
+    }
+}
+
+#[async_trait::async_trait]
+impl<S> ServeHttp for SetStrategyEndpoint<S>
+where
+    S: Strategy + 'static + DeserializeOwned,
+{
+    async fn response(&self, http_session: &mut ServerSession) -> Response<Vec<u8>> {
+        info!("request received");
+        match http_session {
+            ServerSession::H1(session) => {
+                let request_header = session.req_header();
+                if request_header.method != Method::POST {
+                    return Response::builder()
+                        .status(StatusCode::METHOD_NOT_ALLOWED)
+                        .body(Vec::new())
+                        .unwrap();
```

Typically I wouldn't `unwrap` here, I'd return an Error 500 so that the service can keep running but log a useful diagnostic.  Unit or canary tests would pick this up then.